### PR TITLE
feat: use gpt-4o mini in eval tests

### PR DESCRIPTION
**Pull request by friedrichwilken**
**Description**

Changes proposed in this pull request:

- use `gpt-4.o mini` in the evaluation tests

**Related issue(s)**

### CODE REVIEW

**Comment by friedrichwilken** [comment]
[here](https://github.com/kyma-project/kyma-companion/actions/runs/11342613117/job/31543679881?pr=213) ist the last eval test run, which shows that it is in general still operational. I will remove the requirement from the PR because it will always fail due to rate limits.

File: .github/workflows/pull-evaluation-tests.yaml [hunk 1]
```diff
@@ -200,6 +200,8 @@ jobs:
           echo "AICORE_AUTH_URL=$(yq eval '.stringData.auth_url' aicore-secret.yaml)" >> $GITHUB_OUTPUT
           echo "AICORE_BASE_URL=$(yq eval '.stringData.base_url' aicore-secret.yaml)" >> $GITHUB_OUTPUT
           echo "AICORE_RESOURCE_GROUP=$(yq eval '.stringData.resource_group' aicore-secret.yaml)" >> $GITHUB_OUTPUT
+          echo "AICORE_CONFIGURATION_ID_GPT4_MINI=$(yq eval '.stringData.configuration_id_gpt4_mini' aicore-secret.yaml)" >> $GITHUB_OUTPUT
+          echo "AICORE_DEPLOYMENT_ID_GPT4_MINI=$(yq eval '.stringData.deployment_id_gpt_mini4' aicore-secret.yaml)" >> $GITHUB_OUTPUT
           echo "AICORE_CONFIGURATION_ID_GPT4=$(yq eval '.stringData.configuration_id_gpt4' aicore-secret.yaml)" >> $GITHUB_OUTPUT
```
Comment (grischperl) [inline .github/workflows/pull-evaluation-tests.yaml:205]: can these be removed if mini is used now?

File: .github/workflows/pull-evaluation-tests.yaml [hunk 2]
```diff
@@ -221,6 +223,8 @@ jobs:
           AICORE_AUTH_URL: ${{ steps.eval-tests-configs.outputs.AICORE_AUTH_URL }}
           AICORE_BASE_URL: ${{ steps.eval-tests-configs.outputs.AICORE_BASE_URL }}
           AICORE_RESOURCE_GROUP: ${{ steps.eval-tests-configs.outputs.AICORE_RESOURCE_GROUP }}
+          AICORE_CONFIGURATION_ID_GPT4: ${{ steps.eval-tests-configs.outputs.AICORE_CONFIGURATION_ID_GPT4_MINI }}
+          AICORE_DEPLOYMENT_ID_GPT4: ${{ steps.eval-tests-configs.outputs.AICORE_DEPLOYMENT_ID_GPT4_MINI }}
           AICORE_CONFIGURATION_ID_GPT4: ${{ steps.eval-tests-configs.outputs.AICORE_CONFIGURATION_ID_GPT4 }}
           AICORE_DEPLOYMENT_ID_GPT4: ${{ steps.eval-tests-configs.outputs.AICORE_DEPLOYMENT_ID_GPT4 }}
```
Comment (grischperl) [inline .github/workflows/pull-evaluation-tests.yaml:229]: can these be removed if mini is used now?

File: .github/workflows/pull-evaluation-tests.yaml [hunk 3]
```diff
@@ -221,6 +223,8 @@ jobs:
           AICORE_AUTH_URL: ${{ steps.eval-tests-configs.outputs.AICORE_AUTH_URL }}
           AICORE_BASE_URL: ${{ steps.eval-tests-configs.outputs.AICORE_BASE_URL }}
           AICORE_RESOURCE_GROUP: ${{ steps.eval-tests-configs.outputs.AICORE_RESOURCE_GROUP }}
+          AICORE_CONFIGURATION_ID_GPT4: ${{ steps.eval-tests-configs.outputs.AICORE_CONFIGURATION_ID_GPT4_MINI }}
```
Comment (grischperl) [inline .github/workflows/pull-evaluation-tests.yaml:226]: should this be called `...GPT4_MINI`? same thing in the next line?

File: tests/blackbox/src/common/config.py [hunk 4]
```diff
@@ -39,7 +41,9 @@ def __init__(self) -> None:
         self.test_cluster_auth_token = get_env("TEST_CLUSTER_AUTH_TOKEN")
         self.aicore_deployment_id_gpt4 = get_env("AICORE_DEPLOYMENT_ID_GPT4")
```
Comment (grischperl) [inline tests/blackbox/src/common/config.py:42]: can this be cleaned up?

File: tests/blackbox/src/common/config.py [hunk 5]
```diff
@@ -21,6 +21,8 @@ class Config:
     test_cluster_auth_token: str  # Gardener test cluster authentication token.
     aicore_deployment_id_gpt4: str
```
Comment (grischperl) [inline tests/blackbox/src/common/config.py:22]: can this be cleaned up?

File: .github/workflows/pull-evaluation-tests.yaml (hunk 3 above)
Comment (friedrichwilken) [reply .github/workflows/pull-evaluation-tests.yaml:226]: yes

File: .github/workflows/pull-evaluation-tests.yaml (hunk 2 above)
Comment (friedrichwilken) [reply .github/workflows/pull-evaluation-tests.yaml:229]: I think so.

File: tests/blackbox/src/common/config.py (hunk 5 above)
Comment (friedrichwilken) [reply tests/blackbox/src/common/config.py:22]: Let us keep it here so we can rather easily switch between models.

File: .github/workflows/pull-evaluation-tests.yaml (hunk 1 above)
Comment (friedrichwilken) [reply .github/workflows/pull-evaluation-tests.yaml]: Yes.

File: tests/blackbox/src/common/config.py (hunk 4 above)
Comment (friedrichwilken) [reply tests/blackbox/src/common/config.py:42]: Let us keep it here so we can rather easily switch between models, whenever we run the evaluation tests locally.

File: .github/workflows/pull-evaluation-tests.yaml (hunk 2 above)
Comment (friedrichwilken) [reply .github/workflows/pull-evaluation-tests.yaml]: Done.

File: tests/blackbox/src/common/config.py (hunk 4 above)
Comment (friedrichwilken) [reply tests/blackbox/src/common/config.py:42]: Done.

**Comment by friedrichwilken** [comment]
#218 will update the docs accordingly.

File: .github/workflows/pull-evaluation-tests.yaml (hunk 1 above)
Comment (friedrichwilken) [reply .github/workflows/pull-evaluation-tests.yaml]: Done

**Review by grischperl — APPROVED** [review]

**Pull request merged**